docs/sandboxing/README.md

```diff
@@ -10,9 +10,95 @@ There is no clear solution to this yet, but it should take heavy inspiration fro
 
 ### Linux
 
-We could use something like flatpak here, though its unlikely, the current most likely solution is to use linux namespaces to work like a docker-like system.
-
-This is mainly solved other then implementation details
+Linux offers different kinds of isolation using namespacing.
+A fairly low-level tool to access these capabilities is `unshare` that allows creating new namespaces for child processes.
+`unshare` is fine, if full-isolation should be achieved, but we definitely need *some* access to the host system, and without integrating with multiple tools, `unshare` by itself is undesirable.
+A popular tool that utilizes the same capabilities as `unshare` is `bubblewrap`, which is most prominently used by Flatpak. It also allows creating new namespaces, just like `unshare` but additionally provides easy ways to bind paths from the host into the mount namespace.
+
+#### Bubblewrap
+
+Using `bubblewrap` is quite simple. It can be configured by running the `bwrap` binary and defining command line arguments.
+As the name suggests it acts as a wrapper, so after specifying bwrap options the main program inside the sandbox can be specified.
+
+##### Base configuration
+
+- `--unshare-all --share-net` - Try to create all kinds of namespaces except for networking
+  - Doing unprivileged networking in userspace is slow, which is why we will NOT sandbox the root networking namespace
+- `--die-with-parent` - Kill sandbox if launcher crashes/exits
+- `--dev /dev --dev-bind-try /dev/dri /dev/dri` - Allow access to DRM devices for hardware accelerated rendering
+- `--proc /proc` - Mount namespaced procfs at /proc
+- `--unsetenv DBUS_SESSION_BUS_ADDRESS` - The sandbox doesn't have access to the dbus daemon anyway
+- Read-only bind mounts
+  - `/etc/drirc`
+  - `/etc/gai.conf`
+  - `/etc/gnutls`
+  - `/etc/hostname`
+  - `/etc/hosts`
+  - `/etc/je_malloc.conf`
+  - `/etc/localtime`
+  - `/etc/machine-id` - TODO: what are consequences of omitting this
+  - `/etc/os-release` - TODO: what are consequences of omitting this
+  - `/etc/resolv.conf`
+  - `/etc/selinux`
+  - `/etc/timezone`
+  - `/usr`
+  - `/bin`
+  - `/sbin`
+  - `/lib`
+  - `/lib32`
+  - `/lib64`
+  - `/sys/class`
+  - `/sys/dev/char`
+  - `/sys/devices/pci????:??` - Possibly multiple?
+  - `/sys/devices/system/cpu`
+  - Path to game assets and libraries
+- Read-write bind mounts
+  - Path to game root (`.minecraft`)
+
+##### Desktop integration
+
+- Read-only bind mounts
+  - `$XDG_RUNTIME_DIR/pulse` - Access to PulseAudio daemon
+  - `$XDG_RUNTIME_DIR/pipewire-?` - Access to PipeWire daemon (`pipewire-0` if unspecified)
+  - `$XAUTHORITY` - X11 access
+  - `/tmp/.X11-unix/X?` - X11 access (`X0` if `DESKTOP=:0`)
```

Comment:
Access to X11 allows a trivial takeover of user account using for example xdotool to send arbitrary key presses like alt+f2 to open the program launcher of the desktop. I would recommend to either force Wayland usage, or use something like Xephyr to run a nested X server, which shouldn't allow sending keys to the outer X server. I don't know if that supports GPU acceleration though.